aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,319 @@
+
+Filters and Attributes
+^^^^^^^^^^^^^^^^^^^^^^
+It's often convenient to implement filter interfaces as *Attributes*. Filter attributes are applied to controllers and action methods. The framework includes built-in attribute-based filters that you can subclass and customize. For example, the following filter inherits from `ResultFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ResultFilterAttribute/index.html>`_, and overrides its ``OnResultExecuting`` method to add a header to the response.
+
+.. _add-header-attribute:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/AddHeaderAttribute.cs
+  :language: c#
+  :emphasize-lines: 5,16
+
+Attributes allow filters to accept arguments, as shown in the example above. You would add this attribute to a controller or action method and specify the name and value of the HTTP header you wished to add to the response:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/SampleController.cs
+  :language: c#
+  :emphasize-lines: 1
+  :lines: 6-12,25
+  :dedent: 4
+
+The result of the ``Index`` action is shown below - the response headers are displayed on the bottom right.
+
+.. image:: filters/_static/add-header.png
+
+Several of the filter interfaces have corresponding attributes that can be used as base classes for custom implementations.
+
+Filter attributes:
+
+- `ActionFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ActionFilterAttribute/index.html>`_
+- `AuthorizationFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/AuthorizationFilterAttribute/index.html>`_
+- `ExceptionFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ExceptionFilterAttribute/index.html>`_
+- `ResultFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ResultFilterAttribute/index.html>`_
+
+Dependency Injection
+^^^^^^^^^^^^^^^^^^^^
+Filters are not directly returned from :doc:`dependency injection </fundamentals/dependency-injection>` (DI). Filters that are implemented as attributes and added directly to controller classes or action methods cannot have constructor dependencies provided by DI. This is because attributes must have their constructor parameters supplied where they are applied. This is a limitation of how attributes work.
```

Comment:
> Filters are not directly returned from :doc:`dependency injection </fundamentals/dependency-injection>` (DI)

They can if you use the ServiceFilterAttribute, right? I think you can remove this sentence.